src/Resources/queries/widget.js

```diff
@@ -0,0 +1,405 @@
+                const $code = $('<code />').html(PhpDebugBar.Widgets.highlight(statement.sql, 'sql')).addClass(csscls('sql'));
+                if (statement.show_copy) {
+                    $code.append(
+                        $('<span title="Copy to clipboard" />')
+                            .addClass(csscls('copy-clipboard'))
+                            .css('cursor', 'pointer')
+                            .on('click', (event) => {
+                                event.stopPropagation();
+                                this.copyToClipboard($code.get(0));
+                            })
+                    );
+                }
+                $li
+                    .attr('data-connection', statement.connection)
+                    .attr('data-duplicate', this.duplicateQueries.has(statement))
+                    .append($code);
+            }
+
+            if (statement.width_percent) {
+                $li.append(
+                    $('<div />').addClass(csscls('bg-measure')).append(
+                        $('<div />').addClass(csscls('value')).css({
+                            left: `${statement.start_percent}%`,
+                            width: `${Math.max(statement.width_percent, 0.01)}%`,
+                        })
+                    )
+                );
+            }
+
+            if ('is_success' in statement && !statement.is_success) {
+                $li.addClass(csscls('error')).append($('<span />').addClass(csscls('error')).text(`[${statement.error_code}] ${statement.error_message}`));
+            }
+            if (statement.duration_str) {
+                $li.append($('<span title="Duration" />').addClass(csscls('duration')).text(statement.duration_str));
+            }
+            if (statement.memory_str) {
+                $li.append($('<span title="Memory usage" />').addClass(csscls('memory')).text(statement.memory_str));
+            }
+            if (statement.connection) {
+                $li.append($('<span title="Connection" />').addClass(csscls('database')).text(statement.connection));
+            }
+            if (statement.xdebug_link) {
+                $li.append(
+                    $('<span title="Filename" />')
+                        .addClass(csscls('filename'))
+                        .text(statement.xdebug_link.line ? `${statement.xdebug_link.filename}#${statement.xdebug_link.line}` : statement.xdebug_link.filename)
+                        .append(
+                            $('<a/>')
+                                .attr('href', statement.xdebug_link.url)
+                                .attr('data-ajax', statement.xdebug_link.ajax)
+                                .addClass(csscls('editor-link'))
+                                .on('click', (event)=> {
+                                    if ($(event.target).attr('data-ajax') === 'true') {
+                                        fetch($(event.target).attr('href'));
+                                    }
```

Comment:
This looks wrong to me, on ajax url the href link would still open, look at [maximebf/php-debugbar#667](https://github.com/maximebf/php-debugbar/pull/667/files) (_event.stopPropagation();, event.preventDefault();_)

Really, there is no possibility of making an upstream to the [maximebf/php-debugbar](https://github.com/maximebf/php-debugbar) package, this file was recently deleted here: #1519, and all the changes were made upstream to the main package, there are fixes that are made here and forgotten there and vice versa

Reply:
If it is to be merged upstream, we would need a default way of adding internal links, like with the OpenHandler interface. The cache refresh button also doesn't have an equivalent upstream.
But signing etc is a bit more difficult without the framework conventions.

I think it is not a problem to have laravel-debugbar use its own implementation and figure out how to backport this later eventually. Maybe we can think of a command system for the OpenHandler interface or something.

The smaller improvements could perhaps be backported.

Reply:
I see, there are a lot of changes, 
the consistency with the other tabs changes, for example all the tabs have the filters at the bottom, except this one

Reply:
I've changed the code, you're right.

But also this is not meant to be upstreamed. Its a Laravel specific widget replacing the generic sql queries one from php-debugbar.